lib/config.js

```diff
@@ -64,30 +64,41 @@ function loadConfig(filePath) {
 }
 
 /**
+ * Loads plugin from node_modules or cache
+ * @param {string} pluginNameWithoutPrefix Name of the plugin without prefix
+ * @returns {Object} Plugin definition
+ */
+function loadPlugin(pluginNameWithoutPrefix) {
```

Comment:
I think we need something in `eslint.js` as well. People can opt-in to environments within inline configurations, and I don't think that will work with just these changes.

We probably need to do something similar to rules, where there's an object that gets loaded and then reused in `eslint.js`.